### DISCUSSION TITLE
part-ii/2020-129/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#71] [Question comments]
Part II 2020 Quantum Information and Computation: Paper 3, Section II, C 

https://questions.tripos.org/part-ii/2020-129/

**Comment by thenewtonium** [comment]
**(a)** - apparently they literally just want you to point out that $f$ is one-to-one resp. two-to-one.

<details> <summary> <b>(b)(iv)</b> - found slightly annoying to write out, even though the idea is not that deep.  What I ended up saying was</summary>

> Let $A \subset B_n$ s.t. each element $x\in A$ has a unique value of $f(x)$, and that $f(A) = f(B_n)$. Then by (2)[^first], $B_n= A \sqcup (A \oplus z)$.

so that I could split up the sum over $B_n$ in the final state of **(b)(i)**.

[^first]: $[f(x)=f(y)] \text { if and only if } x \oplus y \in\left\{0^{n}, z\right\}$
</details>
